src/tarantool.client/NetworkStreamPhysicalConnection.cs

```diff
@@ -52,14 +55,14 @@ public IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallbac
         {
             CheckConnectionStatus();
 
-            return _stream.BeginRead(buffer, offset, count, callback, state);
+            return _disposed ? null : _stream.BeginRead(buffer, offset, count, callback, state);
```

Comment:
By `IDisposable` specification we should throw ObjectDisposedException if object is disposed.